terraform/state_bucket/provider.tf

```diff
@@ -0,0 +1,20 @@
+terraform {
+  required_version = "0.13.0"
+
+  # When bootstrapping this, you will have to manually disable this and set the bucket up first.
+  backend "s3" {
+    bucket         = "osquery-foundation-terraform-state"
+    region         = "eu-west-2"
+    key            = "tf/osquery/aws/backend.tfstate"
+    role_arn       = "arn:aws:iam::TBD:role/osquery_tfdeployer"
+    dynamodb_table = "osquery-foundation-terraform-lock"
+  }
+}
+
+provider "aws" {
```

Comment:
This might be a higher level code organization question, but how do you think it makes sense to split things up? 

Is each subdir in terraform it's own thing? So `state_bucket` creates the state bucket, and then there might be others? Or is there a big thing? And if the latter, I kinda assume there might be multiple aws providers?

Reply:
So my general thought here is to group by function, the terraform bucket is a sole resource with no dependencies so having it at the top level in the folder organisation makes the most sense. 

I also don't wish to tie the backend to any other Code or  Modules where it would be touched.